test/runtests.jl

```diff
@@ -91,6 +91,18 @@ for xtype in [:Float64, :BigFloat, :Int64]
         end
     end
 end
+
+# Treat mod separately because of discontinuities at integers
```

Comment:
I don't think this is the problem, it's not discontinuous at integers in the tests here since we use a floating point number as second argument.

Reply:
It's still discontinuous at multiples of the 2nd argument, if I'm thinking right. For finite-differencing to be safe, the 1st argument ought to be safely away from any discontinuity, which isn't guaranteed with two random numbers. I'd pick something like `13+rand(), rand(3:5)`.

Reply:
I agree that it's not guaranteed, hence I suggested using `rand() + 0.5` or `rand() + 1` instead of `rand()` for the second argument. I think this would make it quite unlikely that the first argument `rand(1:10)` is close to a singularity. I assume currently on Julia 1.7 we are unlucky and the second argument is too close to 0.

Reply:
Please see the plots I posted below.